api/tests/mocha/services/users.spec.js

```diff
@@ -2604,6 +2604,55 @@ describe('Users service', () => {
       ]);
     });
 
+    it('should parse csv, trim spaces and not split strings with commas inside', async () => {
+      /* eslint-disable max-len */
+      const csv = 'password,username,type,place,token_login,contact.name,contact.phone,contact.address\n' +
+        ',mary,person,498a394e-f98b-4e48-8c50-f12aeb018fcc,TRUE,mary,2652527222,"1 King ST, Kent Town, 55555"\n' +
+        'Secret9876,devi,person,498a394e-f98b-4e48-8c50-f12aeb018fcc,truthy mistake,devi,265252,"12 King ST, Kent Town, 55555"\n' +
+        'Secret1144,jeff,person,498a394e-f98b-4e48-8c50-f12aeb018fcc,,jeff,26599102,"27 King ST, Kent Town, 55555"\n' +
+        'Secret5678, peter ,person,498a394e-f98b-4e48-8c50-f12aeb018fcc,FALSE,Peter, 2652279,"15 King ST, Kent Town, 55555 "';
+      /* eslint-enable max-len */
+      sinon.stub(db.medicLogs, 'get').resolves({ progress: {} });
+      sinon.stub(db.medicLogs, 'put').resolves({});
+
+      const result = await service.parseCsv(csv);
+
+      chai.expect(result.users).to.have.deep.members([
+        {
+          password: '',
+          username: 'mary',
+          type: 'person',
+          place: '498a394e-f98b-4e48-8c50-f12aeb018fcc',
+          contact: { name: 'mary', phone: '2652527222', address: '1 King ST, Kent Town, 55555' },
+          token_login: true,
+        },
+        {
+          password: 'Secret9876',
+          username: 'devi',
+          type: 'person',
+          place: '498a394e-f98b-4e48-8c50-f12aeb018fcc',
+          contact: { name: 'devi', phone: '265252', address: '12 King ST, Kent Town, 55555' },
+          token_login: 'truthy mistake',
```

Comment:
What do you think about whenever the attribute is `token_login`, then we prevent other values than boolean, so if a value is invalid then we just leave it undefined? 
`token_login` is kind of reserved, right?

Reply:
This case could only happen if a user goes out of their way to edit the csv exported by the google sheets but I agree with you in that we should validate the input  
I'm not sure this kind of validation should happen in the csv parser though so I'm pushing some changes that will make any truthy value except `true` to be considered `false`. Let me know what you think about it

Reply:
I think your idea makes it safer, thanks!